File: specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/examples-java/FirewallPolicyRuleCollectionGroupList.java
import com.azure.core.util.Context;

/** Samples for FirewallPolicyRuleCollectionGroups List. */
public final class Main {
    /*
     * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2021-08-01/examples/FirewallPolicyRuleCollectionGroupList.json
     */
    /**
     * Sample code: List all FirewallPolicyRuleCollectionGroups for a given FirewallPolicy.
     *
     * @param azure The entry point for accessing resource management APIs in Azure.
     */
    public static void listAllFirewallPolicyRuleCollectionGroupsForAGivenFirewallPolicy(
        com.azure.resourcemanager.AzureResourceManager azure) {
        azure
            .networks()
            .manager()
            .serviceClient()
            .getFirewallPolicyRuleCollectionGroups()
            .list("rg1", "firewallPolicy", Context.NONE);
    }
}

File: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/examples-js/FirewallPolicyRuleCollectionGroupList.js
const { NetworkManagementClient } = require("@azure/arm-network");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to Lists all FirewallPolicyRuleCollectionGroups in a FirewallPolicy resource.
 *
 * @summary Lists all FirewallPolicyRuleCollectionGroups in a FirewallPolicy resource.
 * x-ms-original-file: specification/network/resource-manager/Microsoft.Network/stable/2022-05-01/examples/FirewallPolicyRuleCollectionGroupList.json
 */
async function listAllFirewallPolicyRuleCollectionGroupsForAGivenFirewallPolicy() {
  const subscriptionId = "subid";
  const resourceGroupName = "rg1";
  const firewallPolicyName = "firewallPolicy";
  const credential = new DefaultAzureCredential();
  const client = new NetworkManagementClient(credential, subscriptionId);
  const resArray = new Array();
  for await (let item of client.firewallPolicyRuleCollectionGroups.list(
    resourceGroupName,
    firewallPolicyName
  )) {
    resArray.push(item);
  }
  console.log(resArray);
}

listAllFirewallPolicyRuleCollectionGroupsForAGivenFirewallPolicy().catch(console.error);
